src/RedisSessionStateProvider/RedisSessionStateProvider.cs

```diff
@@ -211,6 +211,7 @@ private SessionStateStoreData GetItemFromSessionStore(bool isWriteLockRequired,
             try
             {
                 SessionStateStoreData sessionStateStoreData = null;
+                LastException = null;
```

Comment:
Instead of resetting `LastException` to `null` (in only one of the public methods), can we just remove the `if (LastException == null)` checks from all methods? What's the danger of executing methods if there was a previous exception?